concordium-consensus/src/Concordium/GlobalState/Persistent/Accounts.hs

```diff
@@ -69,96 +122,138 @@ import Concordium.Types.HashableTo
-      MonadCache (AccountCache (AccountVersionFor pv)) m
+      MonadCache (AccountCache (AccountVersionFor pv)) m,
+      LMDBAccountMap.MonadAccountMapStore m
     )
 
-instance (IsProtocolVersion pv) => Show (Accounts pv) where
-    show a = show (accountTable a)
-
 instance (SupportsPersistentAccount pv m) => MHashableTo m H.Hash (Accounts pv) where
     getHashM Accounts{..} = getHashM accountTable
 
+-- | Get the accounts created for this block or any non-persisted parent block.
+--  Note that this also empties the difference map for this block.
+writeAccountsCreated :: (SupportsPersistentAccount pv m) => Accounts pv -> m ()
+writeAccountsCreated Accounts{..} = do
+    mAccountsCreated <- liftIO $ readIORef accountDiffMapRef
+    case mAccountsCreated of
+        Absent -> return ()
+        Present accountsCreated -> do
+            listOfAccountsCreated <- liftIO $ DiffMap.flatten accountsCreated
+            liftIO $ atomicWriteIORef accountDiffMapRef Absent
+            LMDBAccountMap.insertAccount listOfAccountsCreated
+
+-- Note. We're writing to the LMDB accountmap as part of the 'storeUpdate' implementation below.
+-- This in turn means that no associated metadata is being written to the LMDB database (i.e. the block hash) of the
+-- persisted block. If we need this, then the write to the LMDB database could be done in 'saveBlockState' and the 'DifferenceMap' should be retained up
+-- to that point.
+-- It shouldn't be necessary with this additional metadata as when (potentially) certified blocks are being rolled back, so are the
+-- accounts created in those, in turn this means that the LMDB account map will have entries for all accounts present in the last finalized block.
 instance (SupportsPersistentAccount pv m) => BlobStorable m (Accounts pv) where
     storeUpdate Accounts{..} = do
-        (pMap, accountMap') <- storeUpdate accountMap
         (pTable, accountTable') <- storeUpdate accountTable
         (pRegIdHistory, regIdHistory') <- storeUpdate accountRegIdHistory
         let newAccounts =
                 Accounts
-                    { accountMap = accountMap',
-                      accountTable = accountTable',
-                      accountRegIdHistory = regIdHistory'
+                    { accountTable = accountTable',
+                      accountRegIdHistory = regIdHistory',
+                      ..
                     }
-        return (pMap >> pTable >> pRegIdHistory, newAccounts)
+
+        -- put an empty 'OldMap.PersistentAccountMap'.
+        -- In earlier versions of the node the above mentioned account map was used,
+        -- but this is now superseded by the 'LMDBAccountMap.MonadAccountMapStore'.
+        -- We put this (0 :: Int) here to remain backwards compatible as this simply indicates an empty map.
+        -- This should be revised as part of a future protocol update when the database layout can be changed.
+        return (put (0 :: Word64) >> pTable >> pRegIdHistory, newAccounts)
```

Comment:
Why not just `put OldMap.empty`?

Reply:
It didn't have a `Serialize` instance and I didn't want to add anything to it now it's "deprecated". 

However, I'm `storeUpdate`ing on the empty `OldMap` now, so there's no magic constants.